packages/utilities/src/utilities/StickyHeader/StickyHeader.js

```diff
@@ -290,76 +382,62 @@ class StickyHeader {
      *   with the elements that should be visible starting at the top of the
      *   viewport.
      */
-    let cumulativeOffset = Math.max(
-      Math.min((masthead ? masthead.offsetTop : 0) + oldY - newY, 0),
-      maxScrollaway * -1
+    this._data.cumulativeOffset = Math.max(
+      Math.min(
+        (masthead ? masthead.offsetTop : 0) + oldY - this._data.scrollPos,
+        0
+      ),
+      this._data.maxScrollaway * -1
     );
 
+    /**
+     * Handle each potentially sticky element in the order we expect them to
+     * appear on the page. Important to do this sequentially for
+     * cumulativeOffset to be correctly calculated by the time each of these
+     * methods accesses it.
+     *
+     * @TODO One idea for improving this so the execution order doesn't matter
+     * is to collect our elements into an array ordered by document position,
+     * then loop over that array and execute a corresponding handler method.
+     */
     if (banner) {
-      cumulativeOffset += Math.max(banner.offsetHeight - newY, 0);
+      this._handleBanner();
     }
-
     if (masthead) {
-      masthead.style.transition = 'none';
-      masthead.style.top = `${cumulativeOffset}px`;
-      cumulativeOffset += masthead.offsetHeight;
+      this._handleMasthead();
+    }
+    if (leadspaceSearchBar) {
+      this._handleLeadspaceSearch();
     }
-
     if (tocInner) {
-      tocInner.style.transition = 'none';
-      tocInner.style.top = `${cumulativeOffset}px`;
-
-      tocShouldStick =
-        toc.layout === 'horizontal' || window.innerWidth < gridBreakpoint;
-
-      const tocIsStuck =
-        Math.round(tocInner.getBoundingClientRect().top) <=
-        cumulativeOffset + 1;
-
-      if (tocShouldStick && tocIsStuck) {
-        cumulativeOffset += tocInner.offsetHeight;
-      }
+      this._handleToc();
     }
+  }
 
-    if (!tocInner && leadspaceSearchBar) {
-      const searchShouldBeSticky =
-        leadspaceSearch.getBoundingClientRect().bottom <=
-        leadspaceSearchThreshold;
-      const searchIsSticky = leadspaceSearch.hasAttribute('sticky-search');
-
-      if (searchShouldBeSticky) {
-        if (!searchIsSticky) {
-          leadspaceSearch.style.paddingBottom = `${leadspaceSearchBar.offsetHeight}px`;
-          leadspaceSearch.setAttribute('sticky-search', '');
-          leadspaceSearchInput.setAttribute('large', '');
-
-          window.requestAnimationFrame(() => {
-            leadspaceSearchBar.style.transitionDuration = '110ms';
-            leadspaceSearchBar.style.transform = 'translateY(0)';
-          });
-        }
+  /**
+   * Manages which elements are stuck and where they are positioned. We should
+   * only have one element stuck to the top of the viewport as the page scrolls
+   * down.
+   */
+  _manageStickyElements() {
+    const { localeModal } = StickyHeader.global._elements;
+    const { scrollPos: scrollPosPrevious } = StickyHeader.global._data;
```

Comment:
I'm curious why we're using the singleton here and not just `this`, eg. 
```suggestion
    const { localeModal } = this._elements;
    const { scrollPos: scrollPosPrevious } = this._data;
```
Seems like all the callees, including event handlers, are properly bound to `this` no? Same question goes for other uses of `StickyHeader.global`, which look like all stem from this method. The rest of this method uses `this` directly, so seems like a safe change here to be consistent.